backend/src/public-auth-token/public-auth-token.service.ts

```diff
@@ -55,46 +58,37 @@ export class AuthService {
     user: User,
     identifier: string,
     validUntil: TimestampMillis | undefined,
-  ): [Omit<AuthToken, 'id' | 'createdAt'>, string] {
+  ): [Omit<PublicAuthToken, 'id' | 'createdAt'>, string] {
     const secret = bufferToBase64Url(randomBytes(64));
     const keyId = bufferToBase64Url(randomBytes(8));
     // More about the choice of SHA-512 in the dev docs
     const accessTokenHash = crypto
       .createHash('sha512')
       .update(secret)
       .digest('hex');
-    let token;
     // Tokens can only be valid for a maximum of 2 years
     const maximumTokenValidity =
       new Date().getTime() + 2 * 365 * 24 * 60 * 60 * 1000;
-    if (!validUntil || validUntil === 0 || validUntil > maximumTokenValidity) {
-      token = AuthToken.create(
-        keyId,
-        user,
-        identifier,
-        accessTokenHash,
-        new Date(maximumTokenValidity),
-      );
-    } else {
-      token = AuthToken.create(
-        keyId,
-        user,
-        identifier,
-        accessTokenHash,
-        new Date(validUntil),
-      );
-    }
+    if (!validUntil || validUntil === 0 || validUntil > maximumTokenValidity)
+      validUntil = maximumTokenValidity;
```

Comment:
Do not override params! (Any always use brackets around if bodies)

```suggestion
    const validUntilLimited = !validUntil || validUntil === 0 || validUntil > maximumTokenValidity ? maximumTokenValidity : validUntil;
```